### PR TITLE
Test connecting tenants via password over TLS through proxy

### PR DESCRIPTION
**Pull request by rafiss**
This PR adds infrastructure for testing secure connections through
the multi-tenancy proxy to a standalone SQL server. 

When possible, the ORM configs were updated to support passing in
cluster name and password.

See individual commits.

Closes https://github.com/cockroachdb/cockroach/issues/52413.

### CODE REVIEW

**Comment by rafiss** [comment]
@tbg just added you for awareness since I resurrected your old PR -- feel free to ignore if you don't have anything to comment.

File: testing/main_test.go [hunk 1]
```diff
@@ -368,62 +403,62 @@ FROM
...
+			columnNames: djangoTestColumnNames,
+		}, nothingSkipped(),
+	)
 }
 
 func TestActiveRecord(t *testing.T) {
-	testORM(t, testInfo{language: "ruby", orm: "activerecord"})
+	testORMForAuthModesExcept(t, testInfo{language: "ruby", orm: "activerecord"}, nothingSkipped())
 }
+
```
Comment (otan) [inline testing/main_test.go:464]: is this extra \n from gofmt?
wonder if we even have a linter

**Comment by otan** [comment]
there are only 2 unit tests - is that expected?

**Review by otan — APPROVED** [review]

File: testing/main_test.go (hunk 1 above)
Comment (rafiss) [reply testing/main_test.go]: it was accidental -- no there's no linter on this project :/

**Pull request merged**